src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        int currentQuantity = storage.getOrDefault(transaction.getFruit(), 0);
+        int newQuantity = currentQuantity - transaction.getQuantity();
+        if (newQuantity < 0) {
+            throw new RuntimeException("Insufficient stock for purchase of "
+                    + transaction.getFruit());
+        }
+        storage.put(transaction.getFruit(), newQuantity);
+    }

```

Comment:
Make sure to check the result balance before saving it in the Storage - it should be positive. If the balance is negative, throw a RuntimeException. This implementation correctly throws an exception if the newQuantity is less than zero, adhering to the checklist requirement for handling the Purchase operation.